### DISCUSSION TITLE
Background tasks cancelled

### DISCUSSION THREAD

**Discussion by csuriano23** [encode/starlette#1913] [Potential Issue] [answered]
Hi there,
I've seen some closed bugs in Starlette project (eg. https://github.com/encode/starlette/issues/1438) on the issue of background tasks being cancelled, but I'm not sure if the problem is definitely solved in starlette 0.21.

Anyway I'am experimenting a similar issue.

I run running a starlette background task from a FastAPI/Strawberry server, something like:

```python
                info.context["background_tasks"].add_task(
                    my_task, *my_args
                )
```

The background task performs some AIOHTTP requests on other servers (and it is quite slow in fact). The wrapping client of AIOHTTP is [NIO](https://github.com/poljar/matrix-nio) which I actually use in the background task.

When executing the function inside the request I have no issues, but when calling it as a bg task I receive the famigerous 'asyncio.exceptions.CancelledError'. I've also tried upgrading Starlette to the latest version.

Any clues on that?

**Comment by Kludex** [comment, marked as answer]
This was fixed in Starlette 0.21.0.
